packages/core/data-transfer/src/strapi/providers/remote-destination/index.ts

```diff
@@ -186,6 +171,40 @@ class RemoteStrapiDestinationProvider implements IDestinationProvider {
     });
   }
 
+  async #connectToWebsocket(
+    wsUrl: string,
+    headers?: { Authorization: string },
+    auth?: ITransferTokenAuth
+  ): Promise<WebSocket> {
+    return new Promise((resolve, reject) => {
+      const server = headers ? new WebSocket(wsUrl, { headers }) : new WebSocket(wsUrl);
+      server.once('open', () => {
+        resolve(server);
+      });
+
+      server.once('error', (err) => {
+        if (err.message === 'Unexpected server response: 401') {
```

Comment:
I don't think we should be relying on the message in case that changes. Can we update the server to send an error code and check that?